x/lightclient/keeper/rollback.go

```diff
@@ -37,18 +37,17 @@ func (k Keeper) RollbackCanonicalClient(ctx sdk.Context, rollappId string, newRe
 		return false
 	})
 
-	// clean the optimistic updates valset
-	err := k.PruneSignersAbove(ctx, client, newRevisionHeight-1)
+	// we DO want to prune the signer of the last valid height:
+	// the only reason we didn't do it before was because we were waiting for next validators hash
+	// but now we don't care about that
+	err := k.PruneSignersAbove(ctx, client, lastValidHeight-1)
```

Comment:
this solves the audit issue, in conjunction with the added state root and timestamp validation on the the AfterUpdateState hook:

Now we can safely prune the last valid height, which we know will have an invalid next validators hash, because we know that it was already verified to have a correct state root and timestamp.

Note that the consensus state for lastValidHeight can't be passed in update client after the fork, because it will have the wrong revision.